File: mainActivity/src/main/java/com/sierrawireless/avphone/message/IMessageDisplayer.java
package com.sierrawireless.avphone.message;

import android.text.Spanned;

/**
 * Generic interface for anything that can display success / error messages.
 */
public interface IMessageDisplayer {
    
    void showError(int id, Object... params);

    void showSuccess(int id, Object... params);

    void showErrorMessage(Spanned spanned);
}

File: mainActivity/src/main/java/com/sierrawireless/avphone/message/IMessageDisplayer.kt
package com.sierrawireless.avphone.message

import android.text.Spanned

/**
 * Generic interface for anything that can display success / error messages.
 */
interface IMessageDisplayer {

    fun showError(id: Int, vararg params: Any)

    fun showSuccess(id: Int, vararg params: Any)

    fun showErrorMessage(spanned: Spanned)
}
